### PR TITLE
Update Symbol

### PR DESCRIPTION
**Pull request by sampersand**
This pr updates `Symbol`.

There wasn't actually a ton to do; it was mostly marking things as aliases, and copying signatures over from `String` (eg for `Symbol#[]`).

### CODE REVIEW

File: core/string.rbs [hunk 1]
```diff
@@ -897,10 +897,10 @@ class String
   #     number #=> "9"
   #
   def =~: (Regexp regex) -> Integer?
-        | [T] (_MatchAgainst[self, T] object) -> T
+        | [T] (_MatchAgainst[T] object) -> T
 
-  interface _MatchAgainst[O, T]
```
Comment (sampersand) [inline core/string.rbs:902]: It didnt make much sense to have an `O` field here, as it's only ever being matched against `String`s, so I've removed the field. However, it is a breaking change, so we might want to hold off on this

File: core/symbol.rbs [hunk 2]
```diff
@@ -469,6 +469,4 @@ class Symbol
             | (:ascii | :lithuanian | :turkic) -> Symbol
             | (:lithuanian, :turkic) -> Symbol
             | (:turkic, :lithuanian) -> Symbol
-
-  def clone: (?freeze: true?) -> self
```
Comment (sampersand) [inline core/symbol.rbs:473]: Not actually defined in `Symbol`

File: core/string.rbs (hunk 1 above)
Comment (ParadoxV5) [reply core/string.rbs:906]: Changing from `O = self` to `String` loses type narrowing.

Having a generic `#=~` (with both LHS and RHS type variables) is also fine for #1638.

Comment (sampersand) [reply core/string.rbs]: Resolved by reverting this

File: core/symbol.rbs [hunk 3]
```diff
@@ -152,21 +150,21 @@ class Symbol
   #
   # Related: String#<=>.
   #
-  def <=>: (Symbol other) -> Integer
-         | (untyped other) -> Integer?
+  def <=>: (Symbol object) -> (-1 | 0 | 1)
+         | (untyped) -> nil
```
Comment (soutaro) [inline core/symbol.rbs:154]:
```suggestion
         | (untyped) -> Integer?
```

The overload may be applied to a `Symbol` object too.

File: core/symbol.rbs [hunk 4]
```diff
@@ -240,7 +236,8 @@ class Symbol
   #
   # Related: Symbol#casecmp?, String#casecmp.
   #
-  def casecmp: (untyped other) -> Integer?
+  def casecmp: (Symbol object) -> (-1 | 0 | 1)?
+             | (untyped) -> nil
```
Comment (soutaro) [inline core/symbol.rbs:240]:
```suggestion
             | (untyped) -> Integer?
```

File: core/symbol.rbs [hunk 5]
```diff
@@ -273,7 +270,8 @@ class Symbol
   #
   # Related: Symbol#casecmp, String#casecmp?.
   #
-  def casecmp?: (untyped other) -> bool?
+  def casecmp?: (Symbol object) -> bool?
+              | (untyped) -> nil
```
Comment (soutaro) [inline core/symbol.rbs:274]:
```suggestion
              | (untyped) -> bool?
```

Comment (soutaro) [reply core/symbol.rbs:274]: We can revert the change for this method because they are identical.

**Review by soutaro — CHANGES_REQUESTED** [review]
The `untyped` version should return for the case of the argument is a `Symbol`.

```rb
a = [1, :foo, "bar"].sample     # => a: Integer | Symbol | String | nil

:foo <=> a         # untyped overload is selected, but the actual value may be a Symbol
```

File: core/symbol.rbs [hunk 6]
```diff
@@ -176,7 +174,7 @@ class Symbol
   # variables; see String#=~.
   #
   def =~: (Regexp regex) -> Integer?
-        | [T] (String::_MatchAgainst[self, T] object) -> T
+        | [T] (String::_MatchAgainst[String, T] object) -> T
```
Comment (soutaro) [inline core/symbol.rbs:177]: [Optional] Wondering if you can add a test to confirm the `_MatchAgains` generic interface works correctly with Steep. (in `test/typecheck/_MatchAgainst` or somewhere.)

File: core/symbol.rbs (hunk 3 above)
Comment (ParadoxV5) [reply core/symbol.rbs:154]: * #1875

File: core/symbol.rbs (hunk 6 above)
Comment (sampersand) [reply core/symbol.rbs:177]: Do we have any of these as examples to base it off of?

**Review by soutaro — APPROVED** [review]
👍

**Comment by soutaro** [comment]
@sampersand Added a few tests with 97e5fbc.

**Pull request merged**